zubhub_frontend/zubhub/src/components/multi_step_progress_bar/multiStepProgressBar.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState, useEffect } from 'react'
+import { makeStyles } from '@material-ui/core/styles';
+import { styles } from '../../assets/js/styles/views/create_activity/createActivityStyles'
+import {
+  Grid,
+  Box,
+} from '@material-ui/core';
+import commonStyles from '../../assets/js/styles';
+import { calcAlignLeft, isCompleted, widthToColorPercent, arrayFromStepCount } from './multiStepProgressBarScripts'
+const useStyles = makeStyles(styles);
+const useCommonStyles = makeStyles(commonStyles);
+
+function MultiStepProgressBar(props) {
+  const [state, setState] = useState({...props})
+  useEffect(() => {
+    setState({...props})
+  }, [props])
```

Comment:
Curios why new state is required here? It causes more bugs

Reply:
to keep track of the step variable in the props and update the progressBar accordingly